### PR TITLE
collector extension and python sdk initial commit

### PR DESCRIPTION
**Pull request by wangzlei**
Collector extension and Python sdk layer.
Be aware that:
* Collector in extension is AWS distro OpenTelemetry Collector
* Collector extension contains a default config, export traces to AWS XRay.
* OTel Python dependencies version are not pinged, will update in later PR.
* Python folder contains AWS Lambda function instrumentor, should be moved to OTel Python contrib in near future.
* README for Python layer is not ready.
* CI is not ready.

### CODE REVIEW

File: collector/LICENSE
Comment (anuraaga) [inline collector/LICENSE:1]: Remove LICENSE is already in top of repo

File: collector/config.yaml
Comment (anuraaga) [inline collector/config.yaml:14]: We need to come to a decision on whether it's ok for a repository in `open-telemetry` to host vendor-specific configs, distros, examples. I'm personally not ok with it and would recommend this be in an aws distro repo. Yes it's easier for a user to start if using the aws distros, but I would suggest that in that case this repo doesn't really belong here which has been my general stance. 

@open-telemetry/technical-committee can you advice on this, which originated from

https://github.com/open-telemetry/community/issues/591#issuecomment-800764602
https://github.com/open-telemetry/opentelemetry-lambda/pull/14#discussion_r595603941

There was some concern about TC oversight if in a different repo by @dyladan 

https://github.com/open-telemetry/community/issues/591#issuecomment-756534589

But this doesn't seem to be too big of a deal since instrumentation can still live in open-telemetry, and has already started with https://github.com/open-telemetry/opentelemetry-specification/pull/1442 (assuming it ever gets merged :P). I assume this is the focus of the TC, the instrumentation experience. The actual end user experience is more of a vendor concern IMO - in this case it would be AWS balancing its hat as a cloud vendor and an observability vendor, but possibly then optimizing for an AWS getting-started experience.

File: python/src/otel/otel_sdk/opentelemetry/instrumentation/aws_lambda/__init__.py
Comment (anuraaga) [inline python/src/otel/otel_sdk/opentelemetry/instrumentation/aws_lambda/__init__.py:72]: Are you planning on moving this to the python instrumentation repo similar to the Java version? For @open-telemetry/technical-committee reference, my above comment only holds true if this code is moved to instrumentation (which it should for sure)

File: python/src/otel/otel_sdk/opentelemetry/instrumentation/aws_lambda/__init__.py
Comment (anuraaga) [inline python/src/otel/otel_sdk/opentelemetry/instrumentation/aws_lambda/__init__.py:94]: This is still set even when X-Ray is off or in passive mode, so we need to check whether it's sampled or not like described in https://github.com/open-telemetry/opentelemetry-specification/pull/1442/files. If off, then we need to propagate from the HTTP headers for proxy requests.

File: python/src/otel/otel_sdk/opentelemetry/resource/__init__.py
Comment (anuraaga) [inline python/src/otel/otel_sdk/opentelemetry/resource/__init__.py:8]: Guessing we'd add resource detectors like this and more here

https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main/sdk-extension/opentelemetry-sdk-extension-aws

File: python/src/otel/otel_sdk/otel_wrapper.py
Comment (anuraaga) [inline python/src/otel/otel_sdk/otel_wrapper.py:41]: CI shouldn't leak into the SDK, this should just be using the `OTEL_TRACES_EXPORTER` variable from the spec I think. Though it makes me wonder what this file is for? Doesn't python support the variables natively? We use that support for the java lambda wrappers so no special config wrapper

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/aws-lambda-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambda/v1_0/TracingRequestWrapperBase.java#L23

**Review by anuraaga — CHANGES_REQUESTED** [review]
Blocking this PR until further discussion on the vendor aspect

File: python/src/otel/otel_sdk/opentelemetry/instrumentation/aws_lambda/__init__.py
Comment (anuraaga) [reply python/src/otel/otel_sdk/opentelemetry/instrumentation/aws_lambda/__init__.py:72]: Ah sorry I see in the PR description now that you call this out as TODO, phew

File: python/src/otel/otel_sdk/opentelemetry/instrumentation/aws_lambda/__init__.py
Comment (wangzlei) [reply python/src/otel/otel_sdk/opentelemetry/instrumentation/aws_lambda/__init__.py:94]: yeah, we should add a switcher here to turn off aws trace id propagation.

File: python/src/otel/otel_sdk/opentelemetry/resource/__init__.py
Comment (wangzlei) [reply python/src/otel/otel_sdk/opentelemetry/resource/__init__.py:8]: yes, will move to python contrib with aws instrumentor, if Lambda resource detector is accepted by Python maintainer :)

File: python/src/otel/otel_sdk/otel_wrapper.py
Comment (wangzlei) [reply python/src/otel/otel_sdk/otel_wrapper.py:41]: good.

**Review by codeboten — COMMENTED** [review]
It would be good to split this PR, one for the extension layer and another for the Python SDK

**Comment by wangzlei** [comment]
Split to 2 PRs for collector and python

**Pull request closed without merging**
